src/clj/muuntaja/middleware.clj

```diff
@@ -37,19 +37,17 @@
 (defn wrap-params [handler]
   "Middleware that merges request `:body-params` into `:params`.
   Supports async-ring."
-  (fn
-    ([request]
-     (let [body-params (:body-params request)
-           request (if (map? body-params)
-                     (update request :params merge body-params)
-                     request)]
-       (handler request)))
-    ([request respond raise]
-     (let [body-params (:body-params request)
-           request (if (map? body-params)
-                     (update request :params merge body-params)
-                     request)]
-       (handler request respond raise)))))
+  (letfn [(set-params
+            ([{:keys [params body-params] :as request}]
+             (cond
+               (empty? body-params) request
```

Comment:
test seem to fail. r-m-f had a convention of putting every kind of body into body-params, it can also be a vector, string etc. need to check whether it's a `map?`. See failing tests.